mods/default/functions.lua

```diff
@@ -585,7 +585,7 @@ function default.can_interact_with_node(player, pos)
 
 	-- Is player wielding the right key?
 	local item = player:get_wielded_item()
-	if item:get_name() == "default:key" then
+	if minetest.registered_items[item:get_name()].groups.key == 1 then
```

Comment:
value of `minetest.registered_items` could be nil. please check against that.

Reply:
Do you mean to check for minetest.registered_items[item:get_name()].groups.key is not nil or also minetest.registered_items[item:get_name()] and minetest.registered_items[item:get_name()].groups?

Reply:
`minetest.registered_items[item:get_name()]` and `minetest.registered_items[item:get_name()].groups` (please use variable to substitute former expression) - since both can be nil: Unknown item or there are no groups defined.

Reply:
Changed.